pybossa/view/account.py

```diff
@@ -1170,7 +1176,18 @@ def _add_bookmark(user_name, short_name, bookmark_name, bookmark_url):
     if bookmark_url is None or len(bookmark_url) > MAX_BOOKMARK_URL_LEN:
         raise ValueError('Bookmark URL must be between 1-100 characters.')
 
-    proj_bookmarks[bookmark_name] =  bookmark_url
+    old_bookmark = proj_bookmarks.get(bookmark_name, None)
+    if old_bookmark is not None:
+        created_date = old_bookmark['created']
+    else:
+        created_date = make_timestamp()
+    updated_date = make_timestamp()
+    bookmark_data = {
+        'url': bookmark_url,
+        'created': created_date,
+        'updated': updated_date
+    }
```

Comment:
From ticket description, primary goal of this ticket seems to be ability to sort bookmarks by `create_date` and `updated_date`. As this is backend changes to support this, I think this would need additional methods, data structures to support fast sorting on `create_date` and `updated_date`. Thoughts?

Reply:
Sorting bookmarks by `create_date` and `updated_date` should happen on the frontend. All we would need to do on the backend side is store the dates and pass them to the frontend, which can then handle sorting with a custom sort function. 

So I think the current solution works well for our use case, since we are only dealing with a small amount of bookmarks (< 100) for now.

Reply:
UI would trigger sorting and each such request would be served by backend in order to get latest information, no? Going by this route, I believe backend would need additional data structures and methods for sorting.

Reply:
discussed with Noor. With small dataset, this is fine to sort on frontend. we can revisit in future if need arise to move sorting to backend.